haystack/nodes/image_to_text/transformers.py

```diff
@@ -158,7 +158,15 @@ def generate_captions(
         if len(image_file_paths) == 0:
             raise ImageToTextError("ImageToText needs at least one file path to produce a caption.")
 
-        images_dataset = ListDataset(image_file_paths)
+        # Handling the case where the user passes a String instead of a List as "image_file_paths"
+        image_file_paths2: List[str] = []
+        if type(image_file_paths) is str:
+            # optional: logger.warning("Expected List, got String instead")
+            image_file_paths2.append(str(image_file_paths))
+        else:
+            image_file_paths2 = image_file_paths
+
+        images_dataset = ListDataset(image_file_paths2)
```

Comment:
I think it would be better if we raise an exception in this case. `generate_captions()` is inherited from the `BaseImageToText` class so I'd prefer we keep the type hints unchanged.

If we handle `str` gracefully we'd have to change the type hints to make it clear that it accepts the `str` type.

The problem reported in #5485 is not that we should accept `str` too, is that we must be clear that this method accepts only a list of strings.

Reply:
I have updated the code, now an Error is raised if a string is being passed. Feel free to edit the error message.